src/preferences/dialog/dlgprefmixer.cpp

```diff
@@ -815,9 +835,87 @@ void DlgPrefMixer::slotUpdate() {
     updateMainEQ();
 }
 
-// Draw the crossfader curve graph. Only needs to get drawn when a change
-// has been made.
+void DlgPrefMixer::slotUpdateXFader() {
+    qWarning() << "     .";
+    qWarning() << "     .";
+    qWarning() << "     slotUpdateXFader";
+    // Read values from config only on first update if the xfader curve controls
+    // are still at their default values. This should detect if controller mappings
+    // (or skin attributes) have changed the xfader controls.
+    // Else and on later calls, always read the current state from controls.
+    if (m_initializing &&
+            m_xfCurveCO->get() == m_xfCurveCO->getDefault() &&
+            m_xfCalibrationCO->get() == m_xfCalibrationCO->getDefault() &&
+            m_xfModeCO->get() == m_xfModeCO->getDefault() &&
+            m_xfReverseCO->get() == m_xfReverseCO->getDefault()) {
+        qWarning() << "     --> init from config";
+        m_xFaderCurve = m_pConfig->getValue(kXfaderCurveKey, EngineXfader::kTransformDefault);
+        // "xFaderCalibration" is not stored in the config and it's not expsoed
+        // with a slider here. Each time the slider is touched it's calculated
+        // to get us a smooth curve for ConstPower mode. And hos no effect for
+        // Additive mode.
+        // TODO This also means custom values set by controller mappings are
+        // wiped on shutdown.
+        m_xFaderCal = EngineXfader::getPowerCalibration(m_xFaderCurve);
+        m_xFaderMode = m_pConfig->getValue<int>(kXfaderModeKey);
+        m_xFaderReverse = m_pConfig->getValue<bool>(kXfaderReverseKey);
+    } else {
+        qWarning() << "     --> from controls";
+        // Update xfader from controls
+        // deactivated for now. resolve dupe debug etc.
+        // slotXFaderControlChanged();
+        m_xFaderCurve = m_xfCurveCO->get();
+        m_xFaderCal = m_xfCalibrationCO->get();
+        m_xFaderMode = static_cast<int>(m_xfModeCO->get());
+        m_xFaderReverse = static_cast<bool>(m_xfReverseCO->get());
+    }
+
+    qWarning().noquote() << "         mode:       "
+                         << QString(m_xFaderMode == MIXXX_XFADER_CONSTPWR ? "Const" : "Addi");
+    qWarning() << "         curve:  " << m_xFaderCurve;
+    qWarning() << "         calibration:" << m_xFaderCal;
+    qWarning().noquote() << "         reverse:    "
+                         << QString(m_xFaderReverse ? "Yes" : "normal");
+
+    updateXFaderWidgets();
+}
+
+void DlgPrefMixer::updateXFaderWidgets() {
+    SliderXFader->blockSignals(true);
+    buttonGroupCrossfaderModes->blockSignals(true);
+    checkBoxReverse->blockSignals(true);
```

Comment:
lets use a `QSignalBlocker` instead. Otherwise an exception or an inconsiderate future early return could cause the widgets to stay blocked.